From b34bd60c3b4378ea33768c4dece5b744a7e769b0 Mon Sep 17 00:00:00 2001
From: Harald Hoyer <harald@redhat.com>
Date: Thu, 23 Apr 2015 13:46:52 +0200
Subject: [PATCH] kernel-modules: install all HID drivers

Instead of hardcoding a list of useful drivers, which has to be curated
all the time, just include all HID drivers.

(cherry picked from commit 180e9d78516fb4b2ee5baef44521007a860d4dd2)
---
 modules.d/90kernel-modules/module-setup.sh | 25 ++++++++++++++++++-------
 1 file changed, 18 insertions(+), 7 deletions(-)

diff --git a/modules.d/90kernel-modules/module-setup.sh b/modules.d/90kernel-modules/module-setup.sh
index fba2bc1a..34d85b1c 100755
--- a/modules.d/90kernel-modules/module-setup.sh
+++ b/modules.d/90kernel-modules/module-setup.sh
@@ -40,20 +40,31 @@ installkernel() {
             ehci-hcd ehci-pci ehci-platform \
             ohci-hcd ohci-pci \
             uhci-hcd \
-            xhci-hcd
+            xhci-hcd xhci-pci xhci-plat-hcd
 
-        instmods yenta_socket scsi_dh_rdac scsi_dh_emc \
-            atkbd i8042 usbhid hid-apple hid-sunplus hid-cherry hid-logitech \
-            hid-logitech-dj hid-microsoft firewire-ohci \
-            pcmcia usb_storage nvme hid-hyperv hv-vmbus \
-            sdhci_acpi
+        instmods yenta_socket scsi_dh_rdac scsi_dh_emc scsi_dh_alua \
+                 atkbd i8042 usbhid firewire-ohci pcmcia hv-vmbus \
+                 atkbd i8042 usbhid firewire-ohci pcmcia usb_storage \
+                 nvme hv-vmbus sdhci_acpi
+
+        instmods \
+            "=drivers/hid" \
+            "=drivers/input/serio" \
+            "=drivers/input/keyboard"
 
         if [[ "$(uname -p)" == arm* ]]; then
             # arm specific modules
             hostonly='' instmods \
 	        connector-hdmi connector-dvi encoder-tfp410 \
 	        encoder-tpd12s015 i2c-tegra gpio-regulator \
-		as3722-regulator orion-ehci ehci-tegra 
+		as3722-regulator orion-ehci ehci-tegra
+            instmods \
+                "=drivers/i2c/busses" \
+                "=drivers/regulator" \
+                "=drivers/rtc" \
+                "=drivers/usb/host" \
+                "=drivers/usb/phy" \
+                ${NULL}
         fi
 
         # install virtual machine support
